src/main/scala/is/hail/variant/package.scala

```diff
@@ -29,6 +29,16 @@ package object variant {
           override def nextInt(): Int = it.next().unboxedGT
         }
     }
+
+    def dosageIterator: DoubleIterator = ig match {
+      case gs: GenotypeStream => gs.gsDosageIterator
+      case _ =>
+        new DoubleIterator {
```

Comment:
This also bothers me and isn't the usual interface for `Iterator[T]`. We should have a think about how we could do it differently.